quartz/components/scripts/search.inline.ts

```diff
@@ -248,6 +266,34 @@ document.addEventListener("nav", async (e: unknown) => {
     }
   }
 
+  async function fetchContent(slug: FullSlug) {
+    const targetUrl = new URL(resolveUrl(slug))
+    const contents = await fetch(targetUrl.toString())
+      .then((res) => res.text())
+      .catch((err) => {
+        console.error(err)
+      })
+    const html = p.parseFromString(contents ?? "", "text/html")
+    normalizeRelativeURLs(html, targetUrl)
+    return [...html.getElementsByClassName("popover-hint")]
+  }
+
+  async function displayPreview(slug: FullSlug) {
+    if (!preview) return
+
+    removeAllChildren(preview)
+    const contentDetails = await fetchContent(slug)
+    if (!contentDetails) {
+      if (!preview) return
+      preview.innerHTML = `<p>Could not load preview.</p>`
+    } else {
+      const previewInner = document.createElement("div")
+      previewInner.classList.add("preview-inner")
+      preview.appendChild(previewInner)
```

Comment:
do we ever cleanup the created `preview-inner`? or should we replace the previous child of `preview`

Reply:
not yet I believe. The logic here somewhat similar to how popover works.

Reply:
popover doesn't re-fetch if the div already exists though, let's also do that here

Reply:
will do.

Reply:
Ah here it will just replace the preview-inner.